runtime/src/main/scala/akka/grpc/internal/NettyClientUtils.scala

```diff
@@ -51,20 +52,25 @@ object NettyClientUtils {
       builder = builder.usePlaintext()
     else {
       builder = builder.negotiationType(NegotiationType.TLS)
-      builder = (settings.trustManager, settings.sslProvider) match {
-        case (None, None) =>
-          builder
-        case (tm, provider) =>
-          val context = provider match {
-            case None =>
-              GrpcSslContexts.configure(SslContextBuilder.forClient())
-            case Some(sslProvider) =>
-              GrpcSslContexts.configure(SslContextBuilder.forClient(), sslProvider)
+      builder = settings.sslContext match {
+        case Some(sslContext) =>
+          builder.sslContext(createNettySslContext(sslContext))
+        case None =>
+          (settings.trustManager, settings.sslProvider) match {
+            case (None, None) =>
+              builder
+            case (tm, provider) =>
+              val context = provider match {
+                case None =>
+                  GrpcSslContexts.configure(SslContextBuilder.forClient())
+                case Some(sslProvider) =>
+                  GrpcSslContexts.configure(SslContextBuilder.forClient(), sslProvider)
+              }
+              builder.sslContext((tm match {
+                case None               => context
+                case Some(trustManager) => context.trustManager(trustManager)
+              }).build())
```

Comment:
Hm, this gives precedence to `SSLContext` over `TrustManager`. I _naïvely_ expected the opposite would be preferred but I have no argument backing the idea.

Reply:
I'll throw an error when both are set.